appcenter-data/ios/AppCenterReactNativeData/AppCenterReactNativeData.m

```diff
@@ -50,24 +61,102 @@ + (void)register {
+                  document: (NSDictionary *)documentMap
+                  writeOptions:(NSDictionary *)writeOptionsMap
+                  resolver:(RCTPromiseResolveBlock)resolve
+                  rejecter:(RCTPromiseRejectBlock)reject) {
+    MSWriteOptions *writeOptions = [self getWriteOptions:writeOptionsMap];
+    MSDictionaryDocument *document = [[MSDictionaryDocument alloc] initFromDictionary:documentMap];
+    [MSData createDocumentWithID:documentID
+                        document:document
+                       partition:partition
+                    writeOptions:writeOptions
+               completionHandler:[self dataCompletionHandler:@"Failed create" resolve:resolve reject:reject]];
+}
+
+RCT_EXPORT_METHOD(replace:(NSString *)documentID
+                  partition:(NSString *)partition
+                  document:(NSDictionary *)documentMap
+                  writeOptions:(NSDictionary *)writeOptionsMap
+                  resolver:(RCTPromiseResolveBlock)resolve
+                  rejecter:(RCTPromiseRejectBlock)reject) {
+    MSWriteOptions *writeOptions = [self getWriteOptions:writeOptionsMap];
+    MSDictionaryDocument *document = [[MSDictionaryDocument alloc] initFromDictionary:documentMap];
+    [MSData replaceDocumentWithID:documentID
+                         document:document
+                        partition:partition
+                     writeOptions:writeOptions
+                completionHandler:[self dataCompletionHandler:@"Failed replace" resolve:resolve reject:reject]];
+}
+
+RCT_EXPORT_METHOD(remove:(NSString *)documentID
+                  partition:(NSString *)partition
+                  writeOptions:(NSDictionary *)writeOptionsMap
+                  resolver:(RCTPromiseResolveBlock)resolve
+                  rejecter:(RCTPromiseRejectBlock)reject) {
+    MSWriteOptions *writeOptions = [self getWriteOptions:writeOptionsMap];
+    [MSData deleteDocumentWithID:documentID
+                       partition:partition
+                    writeOptions:writeOptions
+               completionHandler:[self dataCompletionHandler:@"Failed remove" resolve:resolve reject:reject]];
+}
+
+- (void (^)(MSDocumentWrapper * _Nonnull))dataCompletionHandler:(NSString *)errorCode
+                                                        resolve:(RCTPromiseResolveBlock)resolve
+                                                         reject:(RCTPromiseRejectBlock)reject{
+    return ^(MSDocumentWrapper * _Nonnull documentWrapper) {
+        NSMutableDictionary *jsDocumentWrapper = [[NSMutableDictionary alloc] init];
+        jsDocumentWrapper[kMSjsonValueKey] = documentWrapper.jsonValue;
+        jsDocumentWrapper[kMSPartitionKey] = documentWrapper.partition;
+        jsDocumentWrapper[kMSIDKey] = documentWrapper.documentId;
+        jsDocumentWrapper[kMSETagKey] = documentWrapper.eTag;
+        jsDocumentWrapper[kMSLastUpdatedDateKey] = @([documentWrapper.lastUpdatedDate timeIntervalSince1970] * 1000);
```

Comment:
did you test it does not overflow? I find it odd that we are not using 1000L.

Reply:
oh, I forgot its a double that NSDate returns

Reply:
@guperrot Yeah, it shouldn't overflow. `timeIntervalSince1970` returns interval **in seconds** between the date object and 00:00:00 UTC on 1 January 1970 as a **double** type, hence the yield value is way smaller than max value of **double** in Objective-C.